### PR TITLE
Laboratory work #2, Gellertova Loreleya - 23FPL1

### PR DESCRIPTION
**Pull request by LoraGellertova**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -4,6 +4,12 @@
 Text retrieval with BM25
 """
 # pylint:disable=too-many-arguments, unused-argument
+import string
+from typing import Set
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:8]: remove

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -4,6 +4,12 @@
...
+
+
+# with open("assets/stopwords.txt", "r", encoding="utf-8") as file:
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:11]: rm

File: lab_2_retrieval_w_bm25/main.py [hunk 3]
```diff
@@ -33,6 +49,23 @@ def remove_stopwords(tokens: list[str], stopwords: list[str]) -> list[str] | Non
...
+    if (len(tokens) == 0) or (len(stopwords) == 0):
+        return None
+    else:
+        for elem in stopwords:
+            if not isinstance(elem, str):
+                return None
+
+        tokens_cleared = []
+        for elem in tokens:
+            if not isinstance(elem, str):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:63]: check it earlier

File: lab_2_retrieval_w_bm25/main.py [hunk 4]
```diff
@@ -47,6 +80,22 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
...
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(documents, list):
+        return None
+    if len(documents) == 0:
+        return None
+
+    uniq_words = set()
+    for document in documents:
+        if not isinstance(document, list):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:90]: earlier

File: lab_2_retrieval_w_bm25/main.py [hunk 5]
```diff
@@ -47,6 +80,22 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
...
+            return None
+        if len(document) == 0:
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:92]: the same

File: lab_2_retrieval_w_bm25/main.py [hunk 6]
```diff
@@ -47,6 +80,22 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
...
+            return None
+        for word in document:
+            if not isinstance(word, str):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:95]: the same

File: lab_2_retrieval_w_bm25/main.py [hunk 7]
```diff
@@ -62,6 +111,21 @@ def calculate_tf(vocab: list[str], document_tokens: list[str]) -> dict[str, floa
...
     """
+    if not isinstance(vocab, list) or not isinstance(document_tokens, list):
+        return None
+    if len(vocab) == 0 or len(document_tokens) == 0:
+        return None
+
+    word_freq = {}
+    tf_dict = {}
+    for word in vocab:
+        if not isinstance(word, str):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:122]: the same

File: lab_2_retrieval_w_bm25/main.py [hunk 8]
```diff
@@ -95,13 +159,13 @@ def calculate_tf_idf(tf: dict[str, float], idf: dict[str, float]) -> dict[str, f
...
 
 def calculate_bm25(
-    vocab: list[str],
-    document: list[str],
-    idf_document: dict[str, float],
-    k1: float = 1.5,
-    b: float = 0.75,
-    avg_doc_len: float | None = None,
-    doc_len: int | None = None,
+        vocab: list[str],
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:162]: revert indents

File: lab_2_retrieval_w_bm25/main.py [hunk 9]
```diff
@@ -123,7 +187,7 @@ def calculate_bm25(
 
 
 def rank_documents(
-    indexes: list[dict[str, float]], query: str, stopwords: list[str]
+        indexes: list[dict[str, float]], query: str, stopwords: list[str]
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:190]: revert indents

File: lab_2_retrieval_w_bm25/main.py [hunk 10]
```diff
@@ -141,14 +205,14 @@ def rank_documents(
...
-    idf_document: dict[str, float],
-    alpha: float,
-    k1: float = 1.5,
-    b: float = 0.75,
-    avg_doc_len: float | None = None,
-    doc_len: int | None = None,
+        vocab: list[str],
+        document: list[str],
+        idf_document: dict[str, float],
+        alpha: float,
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:211]: revert indents

File: lab_2_retrieval_w_bm25/start.py [hunk 11]
```diff
@@ -26,7 +27,8 @@ def main() -> None:
             documents.append(file.read())
     with open("assets/stopwords.txt", "r", encoding="utf-8") as file:
         stopwords = file.read().split("\n")
-    result = None
+    result = remove_stopwords(tokenize(documents[str]), stopwords)
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/start.py:30]: what is str??

**Review by EIzharikov — COMMENTED** [review]
Very little progress, it's very sad. Try to raise the score to at least 6. Make CI green, answer&fix all my comments and be ready for Friday. Good luck.

File: lab_2_retrieval_w_bm25/main.py (hunk 1 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:8]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:11]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 5 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:92]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 4 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:90]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:63]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 7 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:122]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 8 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:162]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 9 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:190]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 10 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:211]: done

File: lab_2_retrieval_w_bm25/start.py (hunk 11 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/start.py:30]: I wanted to make it work, but it does not, so it is not final version of start

File: lab_2_retrieval_w_bm25/main.py (hunk 6 above)
Comment (LoraGellertova) [reply lab_2_retrieval_w_bm25/main.py:88]: done

File: lab_1_classify_by_unigrams/settings.json [hunk 12]
```diff
@@ -1,3 +1,3 @@
 {
-  "target_score": 0
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/settings.json:2]: revert

**Comment by EIzharikov** [comment]
Nothing to review. I remind you to make CI green for sucessful getting mark.

**Pull request closed without merging**